cmd/nameset-backfill/main.go

```diff
@@ -0,0 +1,166 @@
+package main
+
+import (
+	"crypto/sha256"
+	"crypto/x509"
+	"flag"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/letsencrypt/boulder/cmd"
+	"github.com/letsencrypt/boulder/core"
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/rpc"
+	"github.com/letsencrypt/boulder/sa"
+
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/cactus/go-statsd-client/statsd"
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/jmhodges/clock"
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/gopkg.in/gorp.v1"
+)
+
+type resultHolder struct {
+	Serial  string
+	Issued  time.Time
+	Expires time.Time
+	DER     []byte
+}
+
+type backfiller struct {
+	sa    core.StorageAuthority
+	dbMap *gorp.DbMap
+	stats statsd.Statter
+	log   *blog.AuditLogger
+	clk   clock.Clock
+}
+
+func new(amqpConf *cmd.AMQPConfig, syslogConf cmd.SyslogConfig, statsdURI, dbURI string) (*backfiller, error) {
+	var stats statsd.Statter
+	var err error
+	stats, log := cmd.StatsAndLogging(cmd.StatsdConfig{Server: statsdURI, Prefix: "Boulder"}, syslogConf)
+	sac, err := rpc.NewStorageAuthorityClient("nameset-backfiller", amqpConf, stats)
+	if err != nil {
+		return nil, err
+	}
+	dbMap, err := sa.NewDbMap(dbURI)
+	if err != nil {
+		return nil, err
+	}
+	return &backfiller{sac, dbMap, stats, log, clock.Default()}, nil
+}
+
+func (b *backfiller) run() error {
+	added := 0
+	defer b.log.Info(fmt.Sprintf("Added %d missing certificate name sets to the fqdnSets table", added))
+	for {
+		results, err := b.findCerts()
+		if err != nil {
+			return err
+		}
+		if len(results) == 0 {
+			break
+		}
+		err = b.processResults(results)
+		if err != nil {
+			return err
+		}
+		added += len(results)
+	}
+	return nil
+}
+
+func (b *backfiller) findCerts() ([]resultHolder, error) {
+	var allResults []resultHolder
+	for {
+		fmt.Println("check")
+		var results []resultHolder
+		_, err := b.dbMap.Select(
+			&results,
+			// idk left outer join instead?
+			`SELECT c.serial, c.issued, c.expires, c.der FROM certificates AS c
+       LEFT JOIN fqdnSets AS ns ON c.serial=ns.serial
+       WHERE ns.serial IS NULL
+       AND c.expires > ?
```

Comment:
I would just leave this out. It's nice to have the table fully-filled, even with data about expired certs.